### PR TITLE
Move AstCallback

### PR DESCRIPTION
**Pull request by cdce8p**
## Description
Followup to https://github.com/PyCQA/pylint/pull/6445#discussion_r856933332

/CC: @DanielNoord

### CODE REVIEW

File: pylint/utils/ast_walker.py [hunk 1]
```diff
@@ -6,16 +6,23 @@
...
 from astroid import nodes
 
-from pylint.typing import AstCallback
-
 if TYPE_CHECKING:
     from pylint.checkers.base_checker import BaseChecker
     from pylint.lint import PyLinter
 
+# Callable parameter type NodeNG not completely correct.
+# Due to contravariance of Callable parameter types,
```
Comment (DanielNoord) [inline pylint/utils/ast_walker.py:19]: We'll need to add this to the spelling dict 😓

File: pylint/utils/ast_walker.py [hunk 2]
```diff
@@ -6,16 +6,23 @@
...
+# it should be a Union of all NodeNG subclasses.
+# However, since the methods are only retrived with
```
Comment (DanielNoord) [inline pylint/utils/ast_walker.py:21]:
```suggestion
# However, since the methods are only retrieved with
```

**Review by DanielNoord — APPROVED** [review]
Thanks for the follow up and teaching me more about `Callable` @cdce8p Much appreciated 😄

**Comment by cdce8p** [comment]
> Thanks for the follow up and teaching me more about `Callable` @cdce8p Much appreciated 😄

I've learned much myself 😅

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**